### PR TITLE
GSAGH-473 Update groups default value from 0 to 1 for members/elements created using GsaGH.

### PR DESCRIPTION
**Pull request by DominikaLos**
[GSA-473]

[GSA-473]: https://ovearup.atlassian.net/browse/GSA-473?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by SandeepArup — COMMENTED** [review]
@DominikaLos @psarras We agreed to look for element also but still getting zero for element. I have attached .gh file in ticket to have a look

File: GsaGH/Parameters/2_Geometry/GsaModelDefaultValues.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿namespace GsaGH.Parameters {
+
+  public struct GsaMemberDefaults {
```
Comment (SandeepArup) [inline GsaGH/Parameters/2_Geometry/GsaModelDefaultValues.cs:3]: @DominikaLos @psarras I feel this is not required. Just look for wrapper class "public class GSAElement" and do all default initilization there else you will endup making changes at multiple places and even for load panel

File: GsaGH/Helpers/GsaAPI/GsaElement.cs [hunk 2]
```diff
@@ -1,32 +1,19 @@
...
     private GSAElement() { }
     public GSAElement(Element element) {
       Element = element;
       IsLoadPanel = false;
+      Element.Group = element.Group == 0 ? 1 : element.Group;
     }
     public GSAElement(LoadPanelElement element) {
       LoadPanelElelment = element;
       IsLoadPanel = true;
+      LoadPanelElelment.Group = element.Group == 0 ? 1 : element.Group;
```
Comment (psarras) [inline GsaGH/Helpers/GsaAPI/GsaElement.cs:16]: typo: LoadPanelElelment

File: GsaGH/Parameters/2_Geometry/GsaElement1d.cs [hunk 3]
```diff
@@ -100,8 +100,7 @@ internal GsaElement1d(KeyValuePair<int, GSAElement> element, IReadOnlyDictionary
       Id = element.Key;
       ApiElement = element.Value;
       if (nodes.Keys.Contains(ApiElement.OrientationNode)) {
-        OrientationNode
-          = new GsaNode(Nodes.Point3dFromNode(nodes[ApiElement.OrientationNode], modelUnit));
+        OrientationNode = new GsaNode(Nodes.Point3dFromNode(nodes[ApiElement.OrientationNode], modelUnit));
       }
```
Comment (psarras) [inline GsaGH/Parameters/2_Geometry/GsaElement1d.cs:104]: is this a method? It repeats further down

File: GsaGH/Parameters/2_Geometry/GsaMember3d.cs [hunk 4]
```diff
@@ -41,6 +41,7 @@ public class GsaMember3d {
     public GsaMember3d() {
       ApiMember = new Member() {
         Type = MemberType.GENERIC_3D,
+        Group = GsaMemberDefaults.GroupValue,
       };
```
Comment (psarras) [inline GsaGH/Parameters/2_Geometry/GsaMember3d.cs:45]: is this repeated? Wrap it in a method

File: GsaGH/Parameters/2_Geometry/GsaMember3d.cs [hunk 5]
```diff
@@ -50,6 +51,7 @@ public GsaMember3d() {
     public GsaMember3d(Mesh mesh) {
       ApiMember = new Member {
         Type = MemberType.GENERIC_3D,
+        Group = GsaMemberDefaults.GroupValue,
       };
```
Comment (psarras) [inline GsaGH/Parameters/2_Geometry/GsaMember3d.cs:55]: with this

File: GsaGH/Parameters/2_Geometry/GsaMember2d.cs [hunk 6]
```diff
@@ -67,6 +68,7 @@ public GsaMember2d(
       Brep brep, List<Curve> includeCurves = null, Point3dList includePoints = null) {
       ApiMember = new Member {
         Type = MemberType.GENERIC_2D,
+        Group = GsaMemberDefaults.GroupValue,
       };
```
Comment (psarras) [inline GsaGH/Parameters/2_Geometry/GsaMember2d.cs:72]: wrap it

File: GsaGH/Parameters/2_Geometry/GsaMember2d.cs [hunk 7]
```diff
@@ -57,6 +57,7 @@ public Angle OrientationAngle {
     public GsaMember2d() {
       ApiMember = new Member() {
         Type = MemberType.GENERIC_2D,
+        Group = GsaMemberDefaults.GroupValue,
       };
```
Comment (psarras) [inline GsaGH/Parameters/2_Geometry/GsaMember2d.cs:61]: wrap it

File: GsaGHTests/3_Components/2_Geometry/CreateElement2dTests.cs [hunk 8]
```diff
@@ -72,6 +72,7 @@ public void CreateComponentTest() {
       TestPoint3d(new Point3d(1, 1, 0), output.Value.Mesh.Vertices[2]);
       TestPoint3d(new Point3d(0, 1, 0), output.Value.Mesh.Vertices[3]);
       Assert.Equal(new Length(14, LengthUnit.Inch), output.Value.Prop2ds[0].Thickness);
+      output.Value.ApiElements.ForEach(x => Assert.Equal(1, x.Group));
     }
```
Comment (psarras) [inline GsaGHTests/3_Components/2_Geometry/CreateElement2dTests.cs:77]:
```csharp
[Fact]
public void GroupsShouldDefaultToOne()
{
    var element = new Element();
    var gsaElement = GSAElement(element);
    Assert.Equal(1, gsaElement.group);
}
[Fact]
public void NonDefaultGroupsShouldNotBeModified()
{
    var element = new Element();
    element.group = 2;
    var gsaElement = GSAElement(element);
    Assert.Equal(2, gsaElement.group);
}


```

**Review by psarras — CHANGES_REQUESTED** [review]

**Review by psarras — COMMENTED** [review]
checked members refactoring, looks good

**Review by psarras — APPROVED** [review]

**Pull request merged**